elyra/pipeline/processor.py

```diff
@@ -187,6 +188,57 @@ def log_pipeline_info(self, pipeline_name: str, action_clause: str, **kwargs):
 
             self.log.info(f"{self._type} '{pipeline_name}'{op_clause} - {action_clause} {duration_clause}")
 
+    @staticmethod
+    def _propagate_operation_inputs_outputs(pipeline: Pipeline, sorted_operations) -> None:
+        """
+        All previous operation outputs should be propagated throughout the pipeline.
+        In order to process this recursively, the current operation's inputs should be combined
+        from its parent's inputs (which, themselves are derived from the outputs of their parent)
+        and its parent's outputs.
+        """
+        for operation in sorted_operations:
+            parent_io = set()  # gathers inputs & outputs relative to parent
+            for parent_operation_id in operation.parent_operations:
+                parent_operation = pipeline.operations[parent_operation_id]
+                if parent_operation.inputs:
+                    parent_io.update(parent_operation.inputs)
+                if parent_operation.outputs:
+                    parent_io.update(parent_operation.outputs)
+
+            if parent_io:
+                parent_io.update(operation.inputs)
+                operation.inputs = parent_io
```

Comment:
Should this get "cast" back as a list - `list(parent_io)`?  This might the cause of the test failure.